applications/zpc/components/zwave/zwave_transports/s2/src/zwave_s2_network.c

```diff
@@ -139,20 +139,22 @@ void zwave_s2_network_init()
   zwave_s2_log_security_keys(SL_LOG_INFO);
 #endif
 
+  uint8_t nls_support = 0;
   uint8_t nls_state = 0;
   zwave_node_id_t node_id = zwave_network_management_get_node_id();
-  sl_status_t status = zwapi_get_node_nls(node_id, &nls_state);
+  sl_status_t status = zwapi_get_node_nls(node_id, &nls_state, &nls_support);
   if (status != SL_STATUS_OK) {
     sl_log_error(LOG_TAG, "Unable to read NLS state for Node ID: %d\n", node_id);
     return;
   }
 
+  sl_log_info(LOG_TAG, "NLS support %s for Node ID: %d\n", nls_support == 1 ? "supported" : "not supported", node_id);
```

Comment:
nitpick: 

```suggestion
  sl_log_info(LOG_TAG, "NLS %s for Node ID: %d\n", nls_support == 1 ? "supported" : "not supported", node_id);
```

Besides the typo, I'd suggest one of the two alternatives:

Do not assume that values other than 1 are false. Instead, use 0 as false and true for values other than 0.

```
  sl_log_info(LOG_TAG, "NLS %s for Node ID: %d\n", nls_support == 0 ? "not supported" : "supported", node_id);
```

Or, to be more strict to the specification, explicitly compare against 1 and 0. And treat anything else as an unknown or error.

Reply:
addressed